ast-visual-studio-extension/CxExtension/Services/ASCAService.cs

```diff
@@ -0,0 +1,490 @@
+                        CxAsca scanResult = await cxWrapper.ScanAscaAsync(
+                            fileSource: tempFilePath,
+                            ascaLatestVersion: false,
+                            agent: "Visual Studio"
+                        );
+
+                        if (scanResult.Error != null)
+                        {
+                            Debug.WriteLine($"ASCA scan failed: {scanResult.Error.Description}");
+                            return;
+                        }
+
+                        Debug.WriteLine("ASCA scan completed successfully.");
+                        await DisplayDiagnosticsAsync(scanResult.ScanDetails, document.FullName);
+                    }
+                    finally
+                    {
+                        // Delete the temporary file
+                        if (File.Exists(tempFilePath))
+                        {
+                            File.Delete(tempFilePath);
+                            Debug.WriteLine($"Temporary file deleted: {tempFilePath}");
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Debug.WriteLine($"Failed to process document: {ex.Message}");
+            }
+        }
+
+        private void ClearAllMarkers()
+        {
+            foreach (var marker in _activeMarkers)
+            {
+                try
+                {
+                    marker.Invalidate(); // Removes the marker
+                }
+                catch (Exception ex)
+                {
+                    Debug.WriteLine($"Failed to invalidate marker: {ex.Message}");
+                }
+            }
+
+            _activeMarkers.Clear(); // Clear the list to avoid references to deleted markers
+        }
+
+
+        private async Task DisplayDiagnosticsAsync(List<CxAscaDetail> scanDetails, string filePath)
+        {
+            if (scanDetails == null || string.IsNullOrEmpty(filePath)) return;
+
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+
+            var document = _dte.Documents.Cast<Document>()
+                .FirstOrDefault(doc => doc.FullName == filePath);
+
+            if (document == null) return;
+
+            // Create if not exists
+            if (_errorListProvider == null)
+            {
+                _errorListProvider = new ErrorListProvider(ServiceProvider.GlobalProvider);
+            }
+
+            // Clear existing tasks in Error List and remove existing markers
+            _errorListProvider.Tasks.Clear();
+            ClearAllMarkers();
+
+            var textManager = ServiceProvider.GlobalProvider.GetService(typeof(SVsTextManager)) as IVsTextManager2;
+            if (textManager == null) return;
+
+            IVsTextView viewEx = null;
+            IVsTextLines buffer = null;
+
+            try
+            {
+                int hr = textManager.GetActiveView2(1, null, 0, out viewEx);
```

Comment:
rename to more indicative names
and add a comment about the meaning of the "0", "1", "null"

Reply:
add